### ISSUE TITLE
[FR] log stack trace in some fatel signal handler

### ISSUE THREAD

**Issue by mirageAlchemy** [odygrd/quill#627]
**Is your feature request related to a problem? Please describe.**
Often in a program that is running online, an unexpected segment fault might happen. In this scenario a full stack trace(and possibly register context) on the thread that triggers the signal could be very useful in assisting trace down the bug.

**Describe the solution you'd like**
A new `enable_stacktrace_printing` in [SignalHandlerOptions](https://quillcpp.readthedocs.io/en/latest/users-api.html#_CPPv420SignalHandlerOptions) would be fine. If the `enable_stacktrace_printing` is true, print the stacktrace by [cpptrace](https://github.com/jeremy-rifkin/cpptrace), `<stacktrace>`, or `libunwind`.

**Additional context**
I can indeed implement the signal handler myself, but in that case I would not be able to use features in [SignalHandlerOptions](https://quillcpp.readthedocs.io/en/latest/users-api.html#_CPPv420SignalHandlerOptions).

**Comment by mirageAlchemy** [comment]
Hi, thanks for the reply and this amazing piece of work!
I understand the concerns:
1. it has to be portable
2. it should be a header only library
3. it better support standard as old as c++20

* while [cpptrace](https://github.com/jeremy-rifkin/cpptrace) is a portable library, it fails at 2
* `std::stacktrace` is a c++23 feature and it might be too hasty to require all users to bump to C++23. Not to mention clang [does not yet support it](https://en.cppreference.com/w/cpp/compiler_support#:~:text=%C2%A0%5BCollapse%5D-,Stacktrace%20library,-P0881R7%0AP2301R1).
* libunwind is not portable

Using core dump is a great one, but sometimes for me the core is too large to even dump to the disk. I might turn to defining the signal handler myself in a less portable way, because I mainly work and run program on linux.

Thanks again and I think I will leave the issue open in case in the future any of the above concern is addressed.

**Comment by odygrd** [comment]
Yes, the signal handler is optional, so you can easily take the existing code and adapt it to capture a stack trace without relying on it. 

I’m fine with keeping this issue open—I might revisit it at some point.

If you have some disk space available, using `gdb` on core dumps generally provides much cleaner and more detailed stack traces. 

Another option is to try a core dump library like [minicoredumper](https://www.linutronix.de/minicoredumper/man/man1/minicoredumper.1.html), though I haven’t personally used it.

If you’re interested in giving `gdb` a try, here’s the Python snippet I use to automatically generate a stack trace from a core dump:

```python
gdb_output = subprocess.run(
    ["gdb", "--batch", "-ex", "bt full", "-ex", "quit", exec_path, latest_core_file],
    capture_output=True, text=True
)

logfile.write("\nStack trace from GDB:\n")
logfile.write(gdb_output.stdout)
logfile.write("\n")
```
